x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -1241,8 +1241,11 @@ private void withRequestBody(XContentBuilder builder, CreateApiKeyRequest create
             for (RoleDescriptor roleDescriptor : createApiKeyRequest.getRoleDescriptors()) {
                 withRoleDescriptor(builder, roleDescriptor);
             }
-            builder.endArray() // role_descriptors
-                .endObject(); // apikey
+            builder.endArray(); // role_descriptors
+            if (createApiKeyRequest.getMetadata() != null && createApiKeyRequest.getMetadata().isEmpty() == false) {
```

Comment:
Staying consistent with how we handle other metadata (e.g. for role descriptors) in audit logs and omitting it if it's empty. This is in contrast to `update` where we do want to track empty metadata since it has different semantics than null metadata (see comment in code).